### DISCUSSION TITLE
Wrapper

### DISCUSSION THREAD

**Discussion by stromhaug-tommy** [Macad3D/Macad3D#35] [General]
Dear Tobias,

When trying to use HLRBRep_HLRToShape it says inaccesssible due to it's protection level.
Is it because it is not included in the config file for the wrappper? Would it be possible to add the class?
BRepAlgoAPI_Defeaturing is not in thge config either,but seems to be available.

Tommy

**Comment by dipts** [comment]
You should be able to add these classes. Instead of HLRBRep_HLRToShape, however, I recommend using or extending the helper class [Macad.Occt.Helper.HLRBRepAlgo](https://github.com/Macad3D/Macad3D/blob/main/Source/Macad.Managed/OcctHelper/HLRBRepAlgo.cpp).

**Reply by stromhaug-tommy** [reply]
Does this mean that i can use this functionality : It is also used in conjunction with the [HLRBRep_HLRToShape](https://dev.opencascade.org/doc/refman/html/class_h_l_r_b_rep___h_l_r_to_shape.html) extraction utilities, which reconstruct a new, simplified shape from a selection of calculation results?

Do you know of any other methods for simplifying complex models i occ ?

**Reply by dipts** [reply]
I would not say that HLR shapes are simpler than the original shapes, on the contrary. The result is a loose collection of edges and open wires. Macad uses this functionality for generating drawings, nothing else.

For the simplification of shapes I think exactly the Defeaturing is meant.
